instrumentation/servlet/src/main/java/brave/servlet/TracingFilter.java

```diff
@@ -0,0 +1,89 @@
+package brave.servlet;
+
+import brave.Span;
+import brave.Tracer;
+import brave.Tracing;
+import brave.http.HttpServerParser;
+import brave.propagation.TraceContext;
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public final class TracingFilter implements Filter {
+  /** Creates a tracing filter with defaults. Use {@link #newBuilder(Tracing)} to customize. */
+  public static TracingFilter create(Tracing tracing) {
+    return newBuilder(tracing).build();
+  }
+
+  public static Builder newBuilder(Tracing tracing) {
+    return new Builder(tracing);
+  }
+
+  public static final class Builder {
```

Comment:
ps I looked at doing something like making a base builder class, but it made generics crazy.. it is more distracting to try and do that (or even autovalue) vs implement directly.